lld/ELF/Driver.cpp

```diff
@@ -3023,7 +3056,6 @@ void LinkerDriver::link(opt::InputArgList &args) {
 
   // Garbage collection and removal of shared symbols from unused shared objects.
   invokeELFT(markLive,);
-  demoteSharedAndLazySymbols();
```

Comment:
Does moving this not have an effect on the other functionality of the function?

Reply:
No effect. demoteSharedAndLazySymbols was here to be clear that it needs to be done after markLive, which handles `--as-needed` DSOs. After markLive, these passes are mostly symbol agnostic. `demoteSharedAndLazySymbols` can be moved very late, even to Writer.cpp.

Hmm, I'm going to make a change to move `demoteSharedAndLazySymbols` to Writer.cpp.

Reply:
Moved `demoteSymbols` in e9b9a1d3202d86d9eb2b49c6463fde0f15f9dc94 and merged two symtab iterations in fc5d815d547e534df8fdb997899e0cffc65b9e35